### DISCUSSION TITLE
Get SocketAdress after firing off a Command

### DISCUSSION THREAD

**Discussion by Taragos** [redis/lettuce#2591] [Q&A]
Hello everyone,

How could I retrieve the actual SocketAdress of the redis-node that I communicate with during a command.

For example, when I retrieve a value from a Redis cluster, I would like to log which node exactly was contacted by my client to troubleshoot connection problems.

I did find a solution by implementing a custom tracer, but I can't use that method in production since I want to use the opentelemetry-javaagent which would conflict with my custom tracer.

Is there any other way to log the SocketAdress of the server?

**Comment by mp911de** [comment]
You can register a `RedisConnectionStateListener` with `RedisClient.addListener(…)`. Generally, logical connections can consist of multiple physical connections (e.g. Master/Replica, Cluster) so there's no API that exposes the underlying connection infrastructure.

If you're looking for observability, there's Tracing available via Micrometer and Brave (see the `io.lettuce.core.tracing` package).
